zerver/lib/actions.py

```diff
@@ -2856,8 +2856,8 @@ def decode_email_address(email):
 # the code pretty ugly, but in this case, it has significant
 # performance impact for loading / for users with large numbers of
 # subscriptions, so it's worth optimizing.
-def gather_subscriptions_helper(user_profile):
-    # type: (UserProfile) -> Tuple[List[Dict[str, Any]], List[Dict[str, Any]], List[Dict[str, Any]]]
+def gather_subscriptions_helper(user_profile, include_subscribers=True):
```

Comment:
The thing is, this code path already has to exclude subscribers for (1) invite-only streams you're not on and (2) zephyr mirroring.  So while I thought about extracting another function, it wouldn't save much complexity.